main.c

```diff
@@ -110,13 +118,17 @@ static void pointer_handle_motion(void *data, struct wl_pointer *wl_pointer,
 			}
 		}
 		break;
-	case WL_POINTER_BUTTON_STATE_PRESSED:
-		seat->has_selection = true;
-		seat->selection.x = min(seat->pressed_x, seat->x);
-		seat->selection.y = min(seat->pressed_y, seat->y);
-		seat->selection.width = abs(seat->x - seat->pressed_x);
-		seat->selection.height = abs(seat->y - seat->pressed_y);
-		break;
+	case WL_POINTER_BUTTON_STATE_PRESSED: {
+			int32_t anchor_x = max(0, seat->anchor_x);
+			int32_t anchor_y = max(0, seat->anchor_y);
+			seat->has_selection = true;
+			seat->selection.x = min(anchor_x, seat->x);
+			seat->selection.y = min(anchor_y, seat->y);
+			seat->selection.width = abs(seat->x - anchor_x);
+			seat->selection.height = abs(seat->y - anchor_y);
+			break;
+
+	}
```

Comment:
I have no idea how to indent this switch/case, tried my best regarding the rest of this file

Reply:
One indent level is enough. Instead of `case thing: { … }` you can do `case thing:; …` (note the semicolon) which doesn't mess up indentation.